lib/docker2aci.go

```diff
@@ -167,24 +166,39 @@ func SquashLayers(images []acirenderer.Image, aciRegistry acirenderer.ACIRegistr
 	}
 
 	squashedFilename := getSquashedFilename(parsedDockerURL)
-	squashedImagePath := path.Join(outputDir, squashedFilename)
+	squashedImagePath := filepath.Join(outputDir, squashedFilename)
 
-	squashedImageFile, err := os.Create(squashedImagePath)
+	squashedTempFile, err := ioutil.TempFile(outputDir, "docker2aci-squashedFile-")
 	if err != nil {
 		return "", err
 	}
-	defer squashedImageFile.Close()
+	defer func() {
+		errClose := squashedTempFile.Close()
+		if err != nil {
+			// remove temp file on error
+			errRemove := os.Remove(squashedTempFile.Name())
+			if errRemove != nil {
+				err = errRemove
```

Comment:
Will this mask the "real" error returned to the caller and/or return an error also if writeSquashedImage/ValidateACI complete successfully but Close() failed?